pkg/cloud/services/loadbalancer/loadbalancer.go

```diff
@@ -658,32 +665,86 @@ func (s *Service) ReconcileLoadBalancerMember(openStackCluster *infrav1.OpenStac
 	return nil
 }
 
-func (s *Service) DeleteLoadBalancer(openStackCluster *infrav1.OpenStackCluster, clusterResourceName string) error {
+func (s *Service) checkIfLbPortsExist(openStackCluster *infrav1.OpenStackCluster) (bool, error) {
```

Comment:
See below: we don't need this any more. In the context of the other fix (waiting until the LB is gone) this results in unnecessary API calls. It also depends on internal Octavia implementation details which aren't part of the API, so it adds additional future failure modes without any benefit. I'd very much prefer we removed this and the new function in `port.go`.